src/schema/schema-builder.ts

```diff
@@ -140,106 +144,121 @@ export function parseProject(project: Project, validationContext: ValidationCont
+    const bodyWithoutComments = stripJsonComments(projectSource.body, {whitespace: true});
+    const parseResult = JSONparse(bodyWithoutComments);
+    const pointers = parseResult.pointers;
 
-            // whitespace: true replaces non-whitespace in comments with spaces so that the sourcemap still matches
-            const bodyWithoutComments = stripJsonComments(projectSource.body, {whitespace: true});
-            const parseResult = JSONparse(bodyWithoutComments);
-            const pointers = parseResult.pointers;
+    for (const key in pointers) {
+        const pointer = pointers[key];
+        jsonPathLocationMap[key] = new MessageLocation(projectSource, pointer.value.pos, pointer.valueEnd.pos);
+    }
 
-            for (const key in pointers) {
-                const pointer = pointers[key];
-                jsonPathLocationMap[key] = new MessageLocation(projectSource, pointer.value.pos, pointer.valueEnd.pos);
-            }
+    const data: PlainObject = parseResult.data;
 
-            data = parseResult.data;
+    // arrays are not forbidden by json-lint
+    if (Array.isArray(data)) {
+        validationContext.addMessage(ValidationMessage.error(`JSON file should define an object (is array)`, new MessageLocation(projectSource, 0, projectSource.body.length)));
+        return undefined;
+    }
 
-            // arrays are not forbidden by json-lint
-            if (Array.isArray(data)) {
-                validationContext.addMessage(ValidationMessage.error(`JSON file should define an object (is array)`, new MessageLocation(projectSource, 0, projectSource.body.length)));
-                return undefined;
-            }
+    return {
+        kind: ParsedProjectSourceBaseKind.OBJECT,
+        namespacePath: getNamespaceFromSourceName(projectSource.name),
+        object: data as PlainObject || {},
+        pathLocationMap: jsonPathLocationMap
+    };
+}
 
-            return {
-                kind: ParsedProjectSourceBaseKind.OBJECT,
-                namespacePath: getNamespaceFromSourceName(projectSource.name),
-                object: data as PlainObject || {},
-                pathLocationMap: jsonPathLocationMap
-            };
-        case SourceType.GRAPHQLS:
-            let document: DocumentNode;
-            try {
-                document = parse(projectSource.toGraphQLSource());
-            } catch (e) {
-                if (e instanceof GraphQLError) {
-                    const message = getMessageFromGraphQLSyntaxError(e);
-                    const location = getGraphQLMessageLocation(e);
-                    validationContext.addMessage(ValidationMessage.error(message, location));
-                    return undefined;
-                }
-                throw e;
-            }
+function parseGraphQLsSource(projectSource: ProjectSource, validationContext: ValidationContext): ParsedGraphQLProjectSource | undefined {
+    if (projectSource.body.trim() === '') {
```

Comment:
Just found the core feature from pr title 👍

Reply:
yeah, the refactor commit is by far bigger than the actual one :D